Handle deprecated sdkconfig option

mmakaay opened the pull request:
# What does this implement/fix?

With ESPHome 2022.12.0 came a new ESP-IDF framework version, in which the sdkconfig option `CONFIG_ESP32_PHY_CALIBRATION_AND_DATA_STORAGE` was deprecated in favour of `CONFIG_ESP_PHY_CALIBRATION_AND_DATA_STORAGE`.

This option is used by the esp32 `ignore_efuse_mac_crc` option to circumvent a piece of code of the framework in which the MAC address of the device is read from efuse. The reason to circumvent it, is that the function that handles the reading operation panics when the CRC (also from ufuse, so immutable) mismatches the MAC address from efuse. Some devices have a wrong CRC, which is why this option was implemented.

The problem with using the deprecated sdkconfig option, is that the panicing code has become active again, resulting in a boot loop after flashing with ESPHome 2022.12.0.

Solution from the PR: use the correct sdkconfig option, based on the framwork version against which the code is compiled.

I tested the change with my Xiaomi Bedside Lamp 2 component.
* Flashing 2022.11.0 to 2022.12.0 -> device goes into boot loop after flashing
* Flashing 2022.11.0 to PR commit -> device works after flashing

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes https://github.com/mmakaay/esphome-xiaomi_bslamp2/issues/104

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [x] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

Commit:
Handle deprecated sdkconfig option

ssieb commented:
Is there any point in testing the version?  If someone deliberately uses an older version, then that's on them.  Also, it's a very small number of devices involved.

jesserockz commented:
Esphome with idf has only supported major version 4 or above...

Commit:
Check for framework version >= 4.4 instead of >= 4

Thanks to @jesserockz, who noted: "You check the major version,
but the sdkconfig was changed in the minor version 4.3.x has old
4.4.x has new"

mmakaay commented:
> Is there any point in testing the version? If someone deliberately uses an older version, then that's on them. Also, it's a very small number of devices involved.

I think that checking the version is the cleanest thing to do here. ESPHome allows users to configure the framework version to use, so it doesn't feel out of bounds to me to do so. It might be a small use case, but without this version test, people that downgrade the framework version might end up with bricked devices. 

In fact, the current hot fix for not letting people brick their lamps, is that I published a config package that forces the use of the old framework. This prevents ESPHome 2022.12.0 from breaking stuff.
When the next version of ESPHome would support only the new sdkconfig option, and people would then flash their new firmware using it, the lamp would again be bricked.

With the version check in place, I can keep the hot fix for now and release a version of my firmware that depends on it.
Once ESPHome has the PR in it, I can create a new version of the firmware, and giver that one a config package that uses the latest version of the framework.
The lock-step release that I have to do, will likely result in some lamps using the old and some lamps using the new framework version, while being compiled with the latest and greatest version of ESPHome.

I'm quickly typing this, because I have to drive home in a few minutes, but I hope I got the point across 😄
